pulser-core/pulser/channels/dmm.py

```diff
@@ -72,26 +89,47 @@ def basis(self) -> Literal["ground-rydberg"]:
     def _undefined_fields(self) -> list[str]:
         optional = [
             "bottom_detuning",
+            "global_bottom_detuning",
             "max_duration",
         ]
         return [field for field in optional if getattr(self, field) is None]
 
-    def validate_pulse(self, pulse: Pulse) -> None:
-        """Checks if a pulse can be executed in this DMM.
+    def validate_pulse(
+        self,
+        pulse: Pulse,
+        detuning_map: DetuningMap = DetuningMap(
+            trap_coordinates=[(0, 0)], weights=[1.0]
+        ),
+    ) -> None:
+        """Checks if a pulse can be executed via this DMM on a DetuningMap.
 
         Args:
             pulse: The pulse to validate.
+            detuning_map: The detuning map on which the pulse is applied
+                (defaults to a detuning map with weight 1.0).
         """
         super().validate_pulse(pulse)
         round_detuning = np.round(pulse.detuning.samples, decimals=6)
+        # Check that detuning is negative
         if np.any(round_detuning > 0):
             raise ValueError("The detuning in a DMM must not be positive.")
+        # Check that detuning on each atom is above bottom_detuning
         if self.bottom_detuning is not None and np.any(
-            round_detuning < self.bottom_detuning
+            np.max(detuning_map.weights) * round_detuning
+            < self.bottom_detuning
         ):
             raise ValueError(
-                "The detuning goes below the bottom detuning "
-                f"of the DMM ({self.bottom_detuning} rad/µs)."
+                "The detunings on some atoms go below the local bottom "
+                f"detuning of the DMM ({self.bottom_detuning} rad/µs)."
+            )
+        # Check that distributed detuning is above global_bottom_detuning
+        if self.global_bottom_detuning is not None and np.any(
```

Comment:
Why do you need the `np.any` here? I thought the sum would yield just a number.

Reply:
There is indeed no need of this np.any ! Thanks for noticing this :)

Reply:
Ah I misread round_detuning is still an array, but we can circumvent the use of an any by taking the minimum value of round_detuning.

Reply:
It would make things a bit clearer. Or you can just add a comment about it.